auction-server/src/auction/service/simulator.rs

```diff
@@ -0,0 +1,337 @@
+            return Ok(accounts.into_iter().map(Some).collect());
+        }
+        let result = self
+            .receiver
+            .get_multiple_accounts_with_commitment(keys, CommitmentConfig::processed())
+            .await?;
+        let mut cache = self.account_cache.write().await;
+        for (key, account) in keys.iter().zip(result.value.iter()) {
+            if let Some(account) = account {
+                cache.insert(*key, (account.clone(), Instant::now()));
+            }
+        }
+        Ok(result.value)
+    }
+
+    async fn resolve_lookup_addresses(
+        &self,
+        transactions: &[VersionedTransaction],
+    ) -> client_error::Result<Vec<Pubkey>> {
+        let mut lookup_table_keys: HashMap<Pubkey, HashSet<u8>> = HashMap::default();
+        transactions
+            .iter()
+            .flat_map(|tx| tx.message.address_table_lookups().unwrap_or_default())
+            .for_each(|x| {
+                lookup_table_keys.entry(x.account_key).or_default().extend(
+                    x.readonly_indexes
+                        .iter()
+                        .chain(x.writable_indexes.iter())
+                        .cloned()
+                        .collect::<Vec<_>>(),
+                )
+            });
+
+        let accs = self
+            .get_multiple_accounts_with_cache(
+                &lookup_table_keys.keys().cloned().collect::<Vec<_>>(),
+            )
+            .await?;
+        let mut results = vec![];
+        for ((lookup_table_address, indexes), lookup_table_account) in
+            lookup_table_keys.iter().zip(accs.iter())
+        {
+            results.push(*lookup_table_address);
+            if let Some(lookup_account) = lookup_table_account {
+                if let Ok(table_data_deserialized) =
+                    AddressLookupTable::deserialize(&lookup_account.data)
+                {
+                    if let Ok(addrs) = table_data_deserialized.lookup(
+                        table_data_deserialized.meta.last_extended_slot + 1,
+                        &indexes.iter().cloned().collect::<Vec<_>>(),
+                        &SlotHashes::default(),
+                    ) {
+                        results.extend(addrs.iter());
+                    }
+                }
+            }
+        }
+        Ok(results)
+    }
+
+    async fn fetch_tx_accounts_via_rpc(
+        &self,
+        transactions: &[VersionedTransaction],
+    ) -> RpcResult<AccountsConfig> {
+        let mut keys = transactions
+            .iter()
+            .flat_map(|tx| tx.message.static_account_keys())
+            .cloned()
+            .collect::<HashSet<_>>();
+
+        keys.extend(self.resolve_lookup_addresses(transactions).await?);
+        let keys = keys.into_iter().collect::<Vec<_>>();
+
+        let accounts_with_context = self
+            .receiver
+            .get_multiple_accounts_with_commitment(&keys, CommitmentConfig::processed())
```

Comment:
fyi this has a limit of 100 accounts per query

Reply:
made it chunked